_1327/user_management/templates/login.html

```diff
@@ -12,18 +12,29 @@
 			{% trans "Please login to see this page." %}
 		</div>
 	{% endif %}
-	<form action="{% url 'login' %}" method="post" class="form-horizontal" role="form">
-	{% bootstrap_form form layout='horizontal' bound_css_class='has-error' %}
-	<input type="hidden" name="next" value="{{ next }}" />
-	{% csrf_token %}
-	{% buttons %}
-	<div class="col-md-offset-2 col-md-4">
-		<button type="submit" class="btn btn-primary">
-			{% trans 'Login' %}
-		</button>
-	</div>
-	{% endbuttons %}
-	</form>
+	{% if openid_login and not request.GET.local_login %}
+		<p>
+			{% trans 'Log in using HPI OpenID Connect.' %}
+		</p>
+		{% buttons %}
+			<a href="{% url 'oidc_authentication_init' %}{% if 'next' in request.GET %}?next={{ request.GET.next }}{% endif %}" class="btn btn-primary">
+				{% trans 'Login' %}
+			</a>
+		{% endbuttons %}
+	{% else %}
+		<form action="{% url 'login' %}" method="post" class="form-horizontal" role="form">
```

Comment:
as this posts to the plain login route, the user is shown the openid login screen instead of a form with errors after entering wrong user credentials

Reply:
yes, I can confirm that problem.

Reply:
i can think of only one use case where this might be a problem: when the admin user wants to login on production and provides wrong credentials. all other users will use open id on production, and in development open id will be deactivated.
so i don't think we need to add additional code for this.

Reply:
you might be right, but this is still some kind of a bug and it should be handled correctly. That means that even an admin user should see the login form and the system should tell him that has been a problem logging him in, because username and password did not match

Reply:
if anyone wants to look into this in the next days - feel welcome. it would require some logic to know which login was used and errors then need to be handled accordingly.
as this pr is blocking our much needed production update and i currently don't want to invest more time into the project than absolutely necessary, i won't implement it myself.

Reply:
see #721 for this issue